### DISCUSSION TITLE
JWT App Type Deprecation

### DISCUSSION THREAD

**Discussion by Ralpharama** [Jericho/ZoomNet#222] [General]
Hi there,

Thanks as ever for this great project.

Given that JWT is being deprecated by Zoom (June, 2023), are there any plans to incorporate the Server-to-Server oAuth type into the project?

Thanks!

Reference via Zoom:

New app type: Server-to-server OAuth Apps - We will be offering a server-to-server OAuth app type for our developers so that you can generate an access token securely without needing user interaction. The token that is generated can be used to enable your private server application to get your account owner to grant permission to use certain scopes.

See the changelog for details. In order for a developer to access this app, the admins need to enable permissions for their role using Role Management.
 
JWT Deprecation - The JWT app type will be completely deprecated as of June 2023. New and current users will have 12 months to migrate their JWT based solutions to the server-to-server OAuth app type.

[JWT Faqs](https://marketplace.zoom.us/docs/guides/build/jwt-app/jwt-faq/)
[Server 2 Server oauth](https://marketplace.zoom.us/docs/guides/build/server-to-server-oauth-app/)

**Comment by Jericho** [comment]
> are there any plans to incorporate the Server-to-Server oAuth type into the project?

Absolutely. In fact, I worked on it over the weekend and published a beta. See #221 for details.

**Reply by Ralpharama** [reply]
Great thanks, I'll subscribe to that thread and follow it there :)

**Reply by Ralpharama** [reply]
I just wanted to say thanks once again - I finally got around to updating this to use the new Server-to-server method and this all works perfectly.

**Reply by Jericho** [reply]
You're very welcome but most importantly, thank you for the kind words. Positive comments like yours are a great source of inspiration.
